server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -402,130 +413,601 @@ describe("Routerlicious", () => {
+
+							// Timeout of 100ms to handle cases where no signal is received
+							setTimeout(() => {
+								client.socket.off("signal", signalListener);
+								resolve(undefined);
+							}, 100);
+						});
+					};
 
-						const firstConnectMessage = await connectToServer(
+					const createClient = async (
+						testId: string,
+						testTenantId: string,
+						testSecret: string,
+						v2: boolean = false,
+					): Promise<Client> => {
+						const socket = webSocketServer.createConnection();
+						const connectMessage = await connectToServer(
 							testId,
 							testTenantId,
 							testSecret,
-							firstSocket,
+							socket,
+							v2,
 						);
+						const clientId = connectMessage.clientId;
+						const client: Client = {
+							socket,
+							clientId,
+							signalCount: 0,
+						};
 
-						const secondSocket = webSocketServer.createConnection();
+						return client;
+					};
+
+					const stringSignalContent = "TestSignal";
+
+					let clients: Client[];
+
+					const numberOfClients = 3; // Change the amount of clients to test with (at leat 2 required)
 
-						await connectToServer(testId, testTenantId, testSecret, secondSocket);
+					assert(numberOfClients > 1, "Test requires at least 2 clients");
+
+					describe("v1 signals", () => {
+						beforeEach(async () => {
+							clients = [];
+							for (let i = 0; i < numberOfClients; i++) {
+								const client = await createClient(testId, testTenantId, testSecret);
+								clients.push(client);
+							}
+							clients.forEach((client) => {
+								client.signalPromise = createSignalPromise(client);
+							});
+						});
+						it("can handle a single broadcast signal", async () => {
+							clients[0].socket.send("submitSignal", clients[0].clientId, [
+								stringSignalContent,
+							]);
+
+							const userSignals = await Promise.all(
+								clients.map((client) => client.signalPromise),
+							);
+
+							clients.forEach((client, index) => {
+								assert.equal(
+									client.signalCount,
+									1,
+									`User ${index + 1} should have received 1 signal`,
+								);
+								assert.equal(
+									userSignals[index].content,
+									stringSignalContent,
+									`User ${index + 1} signal content mismatch`,
+								);
+							});
+						});
+
+						it("can handle signals sent from multiple clients", async () => {
+							clients[0].socket.send("submitSignal", clients[0].clientId, [
+								"first signal",
+							]);
+
+							let userSignals = await Promise.all(
+								clients.map((client) => client.signalPromise),
+							);
+
+							clients.forEach((client, index) => {
+								assert.equal(
+									client.signalCount,
+									1,
+									`User ${index + 1} should have received 1 signal`,
+								);
+								assert.equal(
+									userSignals[index].content,
+									"first signal",
+									`User ${index + 1} signal content mismatch`,
+								);
+							});
```

Comment:
There isn't much value here. This is the prior test case. You can just skip this.
Ideally this setup would not have a big `await` for these before the second signal as the multiple signal handling case we care about is more about several at once.